pkg/api/v1/control-plane/cache/generic_snapshot.go

```diff
@@ -103,17 +110,26 @@ func (s *GenericSnapshot) Consistent() error {
 
 	for _, ref := range required {
 		if resources, ok := s.typedResources[ref.Type]; ok {
-			if _, ok := resources.Items[ref.Name]; ok {
-				return fmt.Errorf("required resource not in snapshot: %s %s", ref.Type, ref.Name)
+			if _, ok := resources.Items[ref.Name]; !ok {
+				return fmt.Errorf("required resource name not in snapshot: %s %s", ref.Type, ref.Name)
 			}
 		} else {
-			return fmt.Errorf("required resource not in snapshot: %s %s", ref.Type, ref.Name)
+			return fmt.Errorf("required resource type not in snapshot: %s %s", ref.Type, ref.Name)
 		}
 	}
 
 	return nil
 }
 
+func (s *GenericSnapshot) MakeConsistent() {
+	// this is fine since generic snapshots are only used by extauth/ratelimit extensions syncers; and those don't
+	// have dependent resources. this will not be called anywhere
+	contextutils.LoggerFrom(context.TODO()).DPanicf("it is an error to call make consistent on a generic snapshot")
```

Comment:
nit: Do we really want to panic? Unless we are catching the panic it blows stuff up and in general we should avoid em.
Perhaps since this truly should never happen its ok to crash out to the top level shutdown loop for a service but I just wanted to make sure we considered this.